tracer/src/Datadog.Trace/Span.cs

```diff
@@ -173,27 +172,28 @@ internal ISpan SetTag(string key, string value)
                     Context.Origin = value;
                     break;
                 case Trace.Tags.SamplingPriority:
-                    if (Enum.TryParse(value, out SamplingPriority samplingPriority) &&
-                        Enum.IsDefined(typeof(SamplingPriority), samplingPriority))
+                    // note: this allows numeric or string representations
+                    // e.g. "AutoKeep" or "1"
+                    if (Enum.TryParse(value, out SamplingPriority samplingPriority))
```

Comment:
Given this should "normally" be an integer, is it preferable to parse it as an integer first, to avoid the expensive `Enum.Parse` path?

Reply:
Good idea, thanks.

Reply:
Done in 3eb370082e5298d61d0525234c7aadbaec24e63f. I benchmarked them, and `int.TryParse()` is always faster than `Enum.TryParse()`, especial for non-numeric strings. `Enum.TryParse()` even allocates heap memory on .NET Framework!